packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -974,6 +974,23 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
         return stats;
     }
 
+    /**
+     * Gracefully stops the current run of the crawler.
+     *
+     * All the tasks active at the time of calling this method will be allowed to finish.
+     */
+    stop(message: string = 'This crawler has been gracefully stopped.') {
```

Comment:
this is trivial to infer. on the other hand, it's a good practice to provide explicit return type on public methods

```suggestion
    stop(message = 'The crawler has been gracefully stopped.'): void {
```